npm/private/npm_translate_lock.bzl

```diff
@@ -442,6 +442,49 @@ def _verify_node_modules_ignored(rctx, importer_paths, bazelignore):
             missing_ignores.append(expected)
     return missing_ignores
 
+def _check_for_conflicting_public_links(npm_imports, public_hoist_packages):
+    all_public_links = {}
+    for (i, _import) in enumerate(npm_imports):
+        for link_package, link_names in _import.link_packages.items():
+            if link_package not in all_public_links:
+                all_public_links[link_package] = {}
+            for link_name in link_names:
```

Comment:
O(n^3) complexity - we should stay vigilant about this becoming slow at some point? it's an "eager fetch" evaluation for ALL bazel users in the workspace. Seems like an easy fix here is just to short-circuit and exit the function at the beginning if `public_hoist_packages` is empty

Reply:
good call; link_packages and link_names should be small in most cases so this should loop should not grow exponentially in large sets of npm deps but still better to not loop if there are no public host packages